Book-A-Desk.Api/HttpHandlers.Reservation.BookADeskErrorHandler.fs

```diff
@@ -0,0 +1,102 @@
+namespace Book_A_Desk.Api
+
+open System
+open Microsoft.AspNetCore.Http
+
+open Book_A_Desk.Domain.Reservation
+
+type BookADeskError =
+    | InvalidEmailAddress
+    | DateLowerThanToday
+    | InvalidOfficeId
+    | OfficeHasNoAvailability of DateTime
+    | UserHadBookedBefore of UserHadBookedBeforeParam
+    | GenericError of string
+
+type ProblemDetailsDto =
+    {
+        Title: string
+        Details: string
+    }
+
+type ResponseError =
+    {
+        StatusCode: int
+        Error: ProblemDetailsDto
+    }
+
+type BookADeskErrorHandler =
+    {
+        MapReservationErrorToAssignBookADeskError: ReservationError -> BookADeskError
+        MapStringToAssignBookADeskError: string -> BookADeskError
+        ConvertErrorToResponseError: BookADeskError -> ResponseError
+    }
+
+module BookADeskErrorHandler =
+    let initialize () =
+
+        let mapReservationErrorToHandlerError = function
+            | ReservationError.InvalidEmailAddress -> InvalidEmailAddress
+            | ReservationError.DateLowerThanToday -> DateLowerThanToday
+            | ReservationError.InvalidOfficeId -> InvalidOfficeId
+            | ReservationError.OfficeHasNoAvailability date -> date |> OfficeHasNoAvailability
+            | ReservationError.UserHadBookedBefore userHadBookedBeforeParam -> userHadBookedBeforeParam |> UserHadBookedBefore
+
+        let mapStringToAssignBookADeskError (description: string) = GenericError description
+
+        let convertErrorToResponseError (error:BookADeskError) =
+            match error with
+            | InvalidEmailAddress ->
```

Comment:
Maybe these errors could be extracted to a specific class to avoid the handler to bloat too much over the time.
Otherwise, we can keep it like that for now it is fine.

Reply:
At this moment, feature's scope only covered Reservation Errors. However, I've covered little more scenarios than were expected. In a near future, another task can work on the rest of possible errors that can happen and we need to analyse and design how to organise each type of error. To my mind, this work would be part of the MVP2.